microimpute/comparisons/data.py

```diff
@@ -352,28 +355,53 @@ def preprocess_data(
         if missing_count > 0:
             logger.warning(f"Data contains {missing_count} missing values")
 
-        # Transform string columns to dummy variables
-        logger.debug("Converting string columns to dummy variables")
+        # Transform boolean and categorical columns to numerical format
+        logger.debug(
+            "Converting boolean and categorical columns to numerical format"
+        )
         try:
-            # Identify string and object columns
+            # Identify boolean columns and convert them to strings
+            bool_columns = [
+                col
+                for col in data.columns
+                if pd.api.types.is_bool_dtype(data[col])
+            ]
+
+            if bool_columns:
+                logger.info(
+                    f"Found {len(bool_columns)} boolean columns to convert: {bool_columns}"
+                )
+                for col in bool_columns:
+                    dummy_info["original_dtypes"][col] = "bool"
+                    # For boolean columns, map the column to itself since we don't create dummies
+                    dummy_info["column_mapping"][col] = [col]
+                    data[col] = data[col].astype(float)
+
+            # Identify string and object columns (excluding already processed booleans)
             string_columns = [
                 col
                 for col in data.columns
-                if pd.api.types.is_string_dtype(data[col])
-                or pd.api.types.is_object_dtype(data[col])
+                if (
+                    pd.api.types.is_string_dtype(data[col])
```

Comment:
For numerical categories, could we:

Parse as category if the unique count < 10 e.g. and warn the user we've done that.